components/crypto/src/utils.rs

```diff
@@ -138,6 +141,121 @@ impl EncryptedKeys {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, Eq, PartialEq)]
+pub struct Keys {
+    pub consensus_pk: PublicKey,
+    pub consensus_sk: SecretKey,
+    pub service_pk: PublicKey,
+    pub service_sk: SecretKey,
+    pub identity_pk: PublicKeyKx,
+    pub identity_sk: SecretKeyKx,
+}
+
+pub fn save_master_key<P: AsRef<Path>, W: AsRef<[u8]>>(
+    path: P,
+    pass_phrase: W,
+    key: Salt,
```

Comment:
Why does `key` have this particular type? Semantically, `key` is not a salt for `pwhash` operation; it is rather a seed for the `SecretTree` (i.e., essentially an input to Blake2b-based KDF [as used in libsodium](https://download.libsodium.org/doc/key_derivation#deriving-keys-from-a-single-high-entropy-key)). Maybe, it could make sense to make a type alias or a newtype for `[u8; secret_tree::SEED_LEN]` in the `secret_tree` crate?

Reply:
Salt is a wrapper around libsodium's `randombytes_buf` which is also used to generate master key via `crypto_kdf_keygen`. But I agree that newtype will be better.